packages/next-app/src/components/SearchPoolForm.tsx

```diff
@@ -33,33 +38,27 @@ export default function SearchPoolForm({ close }: { close?: () => void }) {
     clearErrors,
     resetField,
     formState: { errors },
-  } = useForm<PoolMetadataAttribute>();
-  const { push } = useRouter();
-  const { chain } = useNetwork();
+  } = useForm<PoolAttribute>();
 
   const poolId = watch("poolId");
   const network = watch("network");
 
-  const { data: poolAddress } = pools.gql(network || "1").usePoolAddress(
-    {
-      poolId,
-    },
-    { revalidateIfStale: true }
-  );
-
-  const isPool = poolAddress?.pool?.address ? true : false;
+  const gqlVariables = poolTypeFilter?.length
+    ? { poolId, poolTypes: poolTypeFilter }
+    : { poolId };
+  const { data: poolData } = pools
+    .gql(network || "1")
+    .usePoolsWherePoolTypeInAndId(gqlVariables, { revalidateIfStale: true });
 
-  function handleSubmitForm(formData: PoolMetadataAttribute) {
-    const networkId = formData.network ?? chain?.id.toString();
-    const networkName =
-      networkIdEnumMap[networkId as keyof typeof networkIdEnumMap];
-    push(`/metadata/${networkName}/pool/${formData.poolId}`);
+  const isPool = !!poolData?.pools?.length;
 
+  function handleSubmitForm(formData: PoolAttribute) {
+    onSubmit?.(formData);
```

Comment:
👍 Great addition -- I'd add that if `onSubmit` is not defined this should probably throw an error